### PR TITLE
[docs] Change Icons under Explore APIs section on Home page

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Closes ENG-9006

# How

The Explore APIs section on Home page has outdated icons and uses previously created custom-crafted icons. This PR updates the icons to match our API docs icons (for Image, Notifications, Camera, and Reference) and uses Duotone variants from the `@expo/styleguide-icons`. 

It also removes the previous custom `APIIcons` and replaces Maps with Expo Image.

# Test Plan

Run docs locally and visit: http://localhost:3002/

### Before

CleanShot 2023-09-17 at 13 42 33@2x
CleanShot 2023-09-17 at 13 42 21@2x

### After

CleanShot 2023-09-19 at 01 10 35

CleanShot 2023-09-19 at 01 10 27

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/index.tsx [hunk 1]
```diff
@@ -211,18 +210,26 @@ const Home = () => {
         </Description>
         <CellContainer>
           <Row>
-            <APIGridCell title="Maps" link="/versions/latest/sdk/map-view" icon={<APIMapsIcon />} />
+            <APIGridCell
+              title="Maps"
+              link="/versions/latest/sdk/map-view"
+              icon={<Map01DuotoneIcon className={mergeClasses('h-16 w-16')} />}
```
Comment (Simek) [inline docs/pages/index.tsx:216]:
```suggestion
              icon={<Map01DuotoneIcon className="h-16 w-16" />}
```

Nit, we do not need to use merge utility for icons in this case.

**Review by Simek — APPROVED** [review]
This is looking really good! Thank you for swapping the icons! 👍

File: docs/pages/index.tsx (hunk 1 above)
Comment (amandeepmittal) [reply docs/pages/index.tsx]: Thanks for the tip! I've changed that as per your suggestion for all four icons.

File: docs/pages/index.tsx [hunk 2]
```diff
@@ -1,12 +1,15 @@
 import { css } from '@emotion/react';
-import { Button, theme, typography } from '@expo/styleguide';
+import { Button, DocsLogo, mergeClasses, theme, typography } from '@expo/styleguide';
```
Comment (amandeepmittal) [inline docs/pages/index.tsx:2]:
```suggestion
import { Button, DocsLogo, theme, typography } from '@expo/styleguide';
```

**Review by kbrandwijk — APPROVED** [review]

**Pull request merged**